overlord/snapstate/snapstate.go

```diff
@@ -74,6 +74,19 @@ var ErrNothingToDo = errors.New("nothing to do")
 
 var osutilCheckFreeSpace = osutil.CheckFreeSpace
 
+// TestingLeaveOutKernelUpdateGadgetAssets can be used to simulate an upgrade
+// from a broken snapd that does not generate a "update-gadget-assets" task.
+// See LP:#1940553
```

Comment:
sorry, I should have answered I think the global with the comment would have been enough, we are not in the case of a function that does something. This seems a bit a lot of code for an improbable situation

Reply:
Sure, sorry, I revert this now.

Reply:
well IMHO, I think the function is better, we don't really lose anything by having it (I don't think it's a lot of code, it's merely 4 lines) and it's more defensive to prevent accidental disabling of this critical feature in production